src/test/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheckTest.java

```diff
@@ -435,4 +439,21 @@ public void testVisibilityModifiersOfGenericFields() throws Exception {
         };
         verify(checkConfig, getPath("InputVisibilityModifierGenerics.java"), expected);
     }
+
+    /**
+     * Pitest negates {@code while} loop condition so that it will looks like
+     * while(toVisit == null). In reality there is no chance to pass {@code null}
+     * ast tree to this method, so we have to use reflection.
```

Comment:
it is not ok.
if mutation is like this , so `isStarImport` will always return `false`.
It is not ok that method could return `false` always and that is not noticeable - please recheck tests that do testing of star imports.
Or make a mutation as pitest did and launch checkstyle-tester/diff.groovy to find test case for you.

Reply:
I make a mutation and do regression as you have said, but everything is okay, nothing breaks. This if somehow avoids collisions in short class names for such cases as: java.lang.String and com.sun.org.apache.xpath.internal.operations.String. But I cannot find test case for it.

Reply:
@Nimfadora , after investigation of code.

Reason to test method :
we can not cover this mutation because it force all imports to be non static, but static imports are ignored, so we will not see any affect on validation. We could remove this method at all, and will work correctly as we can not use class with name "*", but in this case internal collection will have short names as "*" that will not make problems, but will be weird in debug. 

TODO: make a UT that call this method by reflection and provide argument artificial AST but correct by structure for "import com.mycompany.*;".

Reply:
done